src/imageops/transform.rs

```diff
@@ -49,16 +49,175 @@ pub fn transpose(buf: &OpBuffer) -> OpBuffer {
   out
 }
 
-/// Rotate an OpBuffer based on the given RawImage's orientation
-pub fn rotate(img: &RawImage, buf: &OpBuffer) -> OpBuffer {
-  match img.orientation.to_flips() {
+fn rotate_buffer(buf: &OpBuffer, orientation: &Orientation) -> OpBuffer {
+  match orientation.to_flips() {
     (false, false, false) => buf.clone(),
     (false, false, true) => flip_vertical(buf),
     (false, true, false) => flip_horizontal(buf),
     (false, true, true) => flip_horizontal(&flip_vertical(buf)),
     (true, false, false) => transpose(buf),
     (true, false, true) => flip_vertical(&transpose(buf)),
     (true, true, false) => flip_horizontal(&transpose(buf)),
-    (true, true, true) => flip_vertical(&transpose(&flip_horizontal(buf))),
+    (true, true, true) => flip_vertical(&flip_horizontal(&transpose(buf))),
```

Comment:
That's good news as it means the new code should be easier to write as transpose is always the first op if it exists.